### PR TITLE
FIX import missing fields on v12

### PR DESCRIPTION
**Pull request by StephaneLesage**
Revert/Cherry-pick import commits from v13/develop apply to v12

### CODE REVIEW

File: htdocs/core/modules/modCommande.class.php [hunk 1]
```diff
@@ -262,6 +263,161 @@ public function __construct($db)
 		$this->export_sql_end[$r] .= ' WHERE c.fk_soc = s.rowid AND c.rowid = cd.fk_commande';
 		$this->export_sql_end[$r] .= ' AND c.entity IN ('.getEntity('commande').')';
 		if (empty($user->rights->societe->client->voir)) $this->export_sql_end[$r] .= ' AND sc.fk_user = '.(empty($user) ? 0 : $user->id);
+		// Imports
```
Comment (eldy) [inline htdocs/core/modules/modCommande.class.php:266]: As this is a new feature, it has been merged for v13 and will be added in v13 only.

Comment (StephaneLesage) [reply htdocs/core/modules/modCommande.class.php]: > As this is a new feature, it has been merged for v13 and will be added in v13 only.

Hi,
do you mean everyhing or just propal/order/invoice ?
OK, so no new features on 12.0.x. Could it be pushed to a hypothetical 12.1 ?

Other fields were exported but just missing on import, so it's not really a new feature.
I modified the PR accordingly.

File: htdocs/core/modules/modProduct.class.php [hunk 2]
```diff
@@ -194,7 +194,7 @@ public function __construct($db)
 		);
 		if (is_object($mysoc) && $mysoc->useNPR()) $this->export_fields_array[$r]['p.recuperableonly'] = 'NPR';
 		if (!empty($conf->fournisseur->enabled) || !empty($conf->margin->enabled)) $this->export_fields_array[$r] = array_merge($this->export_fields_array[$r], array('p.cost_price'=>'CostPrice'));
-		if (!empty($conf->stock->enabled)) $this->export_fields_array[$r] = array_merge($this->export_fields_array[$r], array('p.stock'=>'Stock', 'p.seuil_stock_alerte'=>'StockLimit', 'p.desiredstock'=>'DesiredStock', 'p.pmp'=>'PMPValue'));
+		if (!empty($conf->stock->enabled)) $this->export_fields_array[$r] = array_merge($this->export_fields_array[$r], array('e.ref'=>'DefaultWarehouse', 'p.tobatch'=>'ManageLotSerial', 'p.stock'=>'Stock', 'p.seuil_stock_alerte'=>'StockLimit', 'p.desiredstock'=>'DesiredStock', 'p.pmp'=>'PMPValue'));
```
Comment (eldy) [inline htdocs/core/modules/modProduct.class.php:197]: Adding the warehouse here will make the profile export to not being the profile to export product but to export stock into warehouse (you will get n lines for each warehouse).
There is another dedicated profile for that.

File: htdocs/core/modules/modCommande.class.php (hunk 1 above)
Comment (eldy) [reply htdocs/core/modules/modCommande.class.php]: If 12.1 is 12 + new features, we can But we have just called it 13.0  :-)
If it is adding existing feature not exported, we can say it is a bug.
If it changes the algorithm how we load data, it is better in a new version.

Comment (StephaneLesage) [reply htdocs/core/modules/modCommande.class.php]: > If 12.1 is 12 + new features, we can But we have just called it 13.0 :-)

There is a 10.5 version, but it seems to be the only one like this. I thought minor improvements could be added also to minor revisions of major maintained versions. So OK in x.y.Z you actually don't use Y but only Z for bug fixes.
Then I looked to the history and found [RoadMap](https://wiki.dolibarr.org/index.php?title=Category:RoadMap) and [Release Process](https://wiki.dolibarr.org/index.php?title=FAQ_Release_process)

File: htdocs/core/modules/modProduct.class.php (hunk 2 above)
Comment (StephaneLesage) [reply htdocs/core/modules/modProduct.class.php:197]: What ?
e.ref is warehouse name from llx_entrepot table simply joined by warehouse ID, not the stock !
You already merged my PR on develop back in August => I didn't know develop was for v13 back then.
You make me doubt. I tested it on develop branch and it works fine:
image

**Pull request closed without merging**